src/btree/bt_handle.c

```diff
@@ -524,9 +524,26 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt)
 
     /* If this is the first time opening the tree this run. */
     if (F_ISSET(session, WT_SESSION_IMPORT) || ckpt->run_write_gen < conn->base_write_gen)
-        btree->base_write_gen = btree->run_write_gen = btree->write_gen;
+        btree->run_write_gen = btree->write_gen;
     else
-        btree->base_write_gen = btree->run_write_gen = ckpt->run_write_gen;
+        btree->run_write_gen = ckpt->run_write_gen;
+
+    /*
+     * Don't reset the transaction ids in recovery because rollback to stable in addition to stable
+     * timestamp also depends on transaction ids from the page that are read into cache to decide if
+     * an update needs to be rolled back.
+     *
+     * To achieve that, use the last checkpointed run write generation number as base write
```

Comment:
I don't get the comment. Above it says that you don't want to wipe transaction ids. And here it's saying that we're only retaining transaction ids for pages written after the restart.

Reply:
I will fix the comment now. This was actually from the last changes.